### PR TITLE
feat: add gitguardian hook and update readme

### PR DESCRIPTION
**Pull request by harishv7**
## Problem

While we do have checks post-commit to identify any secrets are committed, we should also attempt to prevent this even before a commit can occur.

After considering some solutions - https://github.com/awslabs/git-secrets and Git Guardian, we decided to go with GitGuardian since we are already using it.

## Solution

We add Git Guardian as a pre-commit hook which scans commits for any leaked secrets before actually performing the commit.

**Breaking Changes**

There is a breaking change for local development since this commit makes Git Guardian setup somewhat mandatory.

- [x] Yes - this PR contains breaking changes
  - Details ...
- [ ] No - this PR is backwards compatible

**New dev dependencies**:

No new dependencies are added to our package. Git Guardian will be installed on the OS-level instead as a brew package.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,33 @@
...
+1. Install GitGuardian
+
+```
+brew install gitguardian/tap/ggshield
+```
+
+2. Add the API Key to your shell startup env (e.g. in `.zshrc`)
+
+```
+export GITGUARDIAN_API_KEY=abc123
```
Comment (QiluXie) [inline README.md:21]: I assume this API key is created by us manually on gitguardian portal, which is fine. But `abc123` is not exactly a good value for an API key, would it be easy to replace it with a more randomized value?

Comment (QiluXie) [inline README.md:21]: nit: is it better to put a comment explaining what value should be put here? 
Something like this:
`export GITGUARDIAN_API_KEY="abc123" //Service API key from gitguardian account`

File: .husky/pre-commit [hunk 2]
```diff
@@ -2,3 +2,4 @@
 . "$(dirname "$0")/_/husky.sh"
 
 npx lint-staged
+ggshield secret scan pre-commit
```
Comment (QiluXie) [inline .husky/pre-commit:5]: nit: it's better to add a newline at the end of the file

Comment (harishv7) [reply .husky/pre-commit]: done

File: README.md (hunk 1 above)
Comment (harishv7) [reply README.md:22]: sure, good idea. done!

**Review by kishore03109 — CHANGES_REQUESTED** [review]
Running ggshield locally creates a`.cache__ggshield` file. By right, since this is a pre-commit hook, there isnt a real need to run ggshield directly. However, if there was a false positive and someone runs ggshield directly, I dont think we should be committing the `.cache__ggshield` to remote. I suggest to defensively add `.cache__ggshield` into our .gitignore, wdyt?

In addition to this, could we add this commit to develop BE + FE vapt + FE develop?

**Comment by harishv7** [comment]
> Running ggshield locally creates a`.cache__ggshield` file. By right, since this is a pre-commit hook, there isnt a real need to run ggshield directly. However, if there was a false positive and someone runs ggshield directly, I dont think we should be committing the `.cache__ggshield` to remote. I suggest to defensively add `.cache__ggshield` into our .gitignore, wdyt?
> 
> In addition to this, could we add this commit to develop BE + FE vapt + FE develop?

@kishore03109 makes sense, added it to gitignore.

As for the FE, yes definitely we will be adding it. Just wanted to address comments on this PR, before I replicate the changes.

As for merging to develop as well, sure we can. But I thought eventually vapt is going into develop. Are we going to merge it soon after vapt?

**Review by kishore03109 — APPROVED** [review]
hmm not sure when we are going to merge vapt into develop, but sure, we can leave it here since it going to dev anyways!

File: .husky/pre-commit (hunk 2 above)
Comment (kishore03109) [inline .husky/pre-commit:5]: Out of curiosity, could we source our env var `GITGUARDIAN_API_KEY` from here? that way we dont need to add anything to our global variable in our machines!

Comment (harishv7) [reply .husky/pre-commit:5]: Sure, I thought by installing globally, devs can use it across all repos and also the global CLI to scan. But I think .env also works

Will update the README and command

**Review by QiluXie — APPROVED** [review]
Thanks @harishv7, can be merged if no comment from @alexanderleegs

**Pull request merged**